From: Giridhar Malavali <giridhar.malavali@cavium.com>
Date: Fri, 13 Oct 2017 15:43:23 -0700
Subject: scsi: qla2xxx: Query FC4 type during RSCN processing
Patch-mainline: v4.15-rc1
Git-commit: 2d57b5efda511886daa984d5bb3da29759bb157c
References: FATE#322910

Based on the FC4 type, login will proceed to either FCP or FC-NVMe
remote ports.

Signed-off-by: Giridhar Malavali <giridhar.malavali@cavium.com>
Signed-off-by: Himanshu Madhani <himanshu.madhani@cavium.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/scsi/qla2xxx/qla_os.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -4761,7 +4761,7 @@ void qla24xx_create_new_sess(struct scsi
 		if (pla)
 			qlt_plogi_ack_unref(vha, pla);
 		else
-			qla24xx_async_gnl(vha, fcport);
+			qla24xx_async_gffid(vha, fcport);
 	}
 
 	if (free_fcport) {
